model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -32,6 +32,16 @@ option java_outer_classname = "RunnerApi";
 import "google/protobuf/any.proto";
 import "google/protobuf/descriptor.proto";
 
+message BeamConstants {
+  enum Constants {
+    // The maximum timestamp for the global window in milliseconds since Jan 1, 1970
+    // Triggers use maxTimestamp to set timers' timestamp. Timers fires when
+    // the watermark passes their timestamps. So, the timestamp needs to be
+    // smaller than the TIMESTAMP_MAX_MILLIS.
+    GLOBAL_WINDOW_MAX_TIMESTAMP_MILLIS = 0 [(beam_constant) = "9223371950454775"];
```

Comment:
This is a seemingly random value; could we comment on where we got this data from?

Reply:
Totally agree it is arbitrary, but wouldn't MAX_TIMESTAMP/2 be arbitrary as well? I suppose it would be more explicitly arbitrary :) I added a comment where this is coming from.

Hex representation does save as some bytes (precisely four).

Reply:
Oh, I was suggesting hex hoping that it'd be something like "FFFFFF...." but, no, `TIMESTAMP_MAX_VALUE` is an odd value as well. (Maybe that should be a constant too?)

I was suggesting `MAX_TIMESTAMP/2` so that it doesn't work when rounding to minutes or hours, but suddenly break when rounding to days (or longer). Looks like there's a `MIN_TIMESTAMP` value as well.